src/cargo/core/compiler/mod.rs

```diff
@@ -1421,8 +1421,26 @@ fn on_stderr_line_inner(
                 rendered: String,
                 message: String,
                 level: String,
-                // `children: Vec<Diagnostic>` if we need to check them recursively
-                children: Vec<Diagnostic>,
+                children: Vec<PartialDiagnostic>,
+            }
+
+            // A partial rustfix::diagnostics::Diagnostic. We deserialize only a
+            // subset of the fields because rustc's output can be extremely
+            // deeply nested JSON in pathological cases involving macro
+            // expansion. Rustfix's Diagnostic struct is recursive containing a
+            // field `children: Vec<Self>`, and it can cause deserialization to
+            // hit serde_json's default recursion limit, or overflow the stack
+            // if we turn that off. Cargo only cares about the 1 field listed
+            // here.
+            #[derive(serde::Deserialize)]
```

Comment:
Sorry for the delay, this slipped through the cracks. 

I don't have any concerns and it makes complete sense why would fix the problem. This may improve performance in some very nested `rustc` output.

You are welcome to put me as the approver if you need